src/System.Runtime.Extensions/src/System/Runtime/Versioning/FrameworkName.cs

```diff
@@ -203,16 +203,17 @@ public FrameworkName(String frameworkName)
             for (int i = 1; i < components.Length; i++)
             {
                 // Get the key/value pair separated by '='
-                string[] keyValuePair = components[i].Split(c_keyValueSeparator);
+                string component = components[i];
+                int separatorIndex = component.IndexOf('=');
 
-                if (keyValuePair.Length != 2)
+                if (separatorIndex == -1 || separatorIndex != component.LastIndexOf('='))
```

Comment:
This is searching more than it needs to, in that it'll keep searching past the first '='.  You can use something like ```component.IndexOf(c_keyValueSeparator, separatorIndex + 1) != -1)```.